src/number_theory/number_field.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright (c) 2021 Ashvni Narayanan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ashvni Narayanan, Anne Baanen
+-/
+
+import algebra.field
+import data.rat.basic
+import ring_theory.algebraic
+import ring_theory.dedekind_domain
+import ring_theory.integral_closure
+import ring_theory.polynomial.rational_root
+
+/-!
+# Number fields
+This file defines a number field and the ring of integers corresponding to it.
+
+## Main definitions
+ - `number_field` defines a number field as a field which has characteristic zero and is finite
+    dimensional over ℚ.
+ - `ring_of_integers` defines the ring of integers (or number ring) corresponding to a number field
+    as the integral closure of ℤ in the number field.
+
+## Implementation notes
+The definitions that involve a field of fractions choose a canonical field of fractions,
+but are independent of that choice.
+
+## References
+* [D. Marcus, *Number Fields*][marcus1977number]
+* [J.W.S. Cassels, A. Frölich, *Algebraic Number Theory*][cassels1967algebraic]
+* [P. Samuel, *Algebraic Theory of Numbers*][samuel1970algebraic]
+
+## Tags
+number field, ring of integers
+-/
+
+/-- A number field is a field which has characteristic zero and is finite
+dimensional over ℚ. -/
+class number_field (K : Type*) [field K] : Prop :=
+[cz : char_zero K]
```

Comment:
```suggestion
[char_zero : char_zero K]
```
Since this is now the instance name

Reply:
`extends` might also be a good choice

Reply:
I'm going to go for fields over `extends`, for reasons of symmetry: because `finite_dimensional` is not a structure, you can't `extend` it.